tests/functionaltests/Tests/NSURLConnectionTests.mm

```diff
@@ -19,6 +19,96 @@
 #import <atomic>
 #import <map>
 #import <random>
+#import <memory>
+
+@interface THRunLoopSpinner : NSObject {
+@public
+    std::shared_ptr<std::atomic<bool>> _shouldStop;
+    BOOL _threadStarted;
+    NSCondition* _threadStartedCondition;
+    NSRunLoop* _loop;
+    pthread_t _thread;
+}
+@property (nonatomic, readonly, retain) NSRunLoop* loop;
+@end
+
+@implementation THRunLoopSpinner
+static void* _helperThreadBody(void* context) {
+    __unsafe_unretained THRunLoopSpinner* unsafeHelper = (__bridge id)context;
+    @autoreleasepool {
+        std::shared_ptr<std::atomic<bool>> pShouldStop;
+        NSRunLoop* loop = nil;
+        { // scope guard for ownership of helper
+            __strong THRunLoopSpinner* helper = unsafeHelper;
+            pShouldStop =
+                helper->_shouldStop; // Capture the shared_ptr; the thread needs to read from it _after the helper is deallocated!_
```

Comment:
nit: would rather the comment be on a separate line if it forces a line break like this